pricegraph/src/orderbook/user.rs

```diff
@@ -62,3 +59,30 @@ impl User {
         self.balances.remove(&token);
     }
 }
+
+fn u256_to_u128_saturating(u256: &U256) -> u128 {
+    if u256.0[2] == 0 && u256.0[3] == 0 {
```

Comment:
ubernit: It might make this easier to understand if we used some higher-level abstractions over `U256`.
```suggestion
    cmp::min(u256, u128::MAX.into()).low_u128()
```

I would make sure that this doesn't incur a performance impact though.